### DISCUSSION TITLE
[Bug] Exessive network request when moving model in workspaces/models view

### DISCUSSION THREAD

**Discussion by MartianInGreen** [open-webui/open-webui#5734] [General]
# Bug Report

## Installation Method

```bash
docker run -d -p 3000:8080 -e OPENAI_API_KEY=your_secret_key -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
``` 

## Environment

- **Open WebUI Version:** v0.3.30
- **Ollama (if applicable):** -/-

- **Operating System:** Ubuntu 22.04 (host), NixOS (client)
- **Browser (if applicable):** Zen 1.0.1-a.5 (Firefox)

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [ ] I have included the Docker container logs. N/A
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Send one network request to update position like is done for the "Move to Top" operation.

## Actual Behavior:

Sends one network request for each model in the model list, regardless of if it's position got updated or not.

## Description

**Bug Summary:**
While moving / re-ordering a model in the Workspace/Models view an excessive amount of network requests gets send each and every time you move a model leading to de-sync and other issues. 

## Reproduction Details

**Steps to Reproduce:**
1. Go to the model page
2. Move a model
3. Observe network requests

## Logs and Screenshots

**Screenshots/Screen Recordings (if applicable):**
![image](https://github.com/user-attachments/assets/15b639c2-05a3-4e31-9f5b-cc2c8bcf31af)
All 202 request are from moving one model, in this case Gemini 1.0 Pro down one space.

**Comment by tjbck** [comment]
Not a bug, however if you find a way to optimise this feel free to make a PR!

**Reply by MartianInGreen** [reply]
I think this definitely is a bug, if you move another model while the network request from the first move are not yet done the second move doesn't happen but there is no feedback to the user that it failed
